src/goose/cli/main.py

```diff
@@ -97,6 +97,43 @@ def list_toolkits() -> None:
         print(f" - [bold]{toolkit_name}[/bold]: {first_line_of_doc}")
 
 
+@goose_cli.group()
+def providers() -> None:
+    """Manage providers"""
+    pass
+
+
+@providers.command(name="list")
+def list_providers() -> None:
+    providers = load_plugins(group="exchange.provider")
+
+    def extract_envs_from_doc(lines: List[str]) -> List[str]:
+        required_envs = []
+        env_block = False
+        for line in lines:
+            if "Required env vars:" in line:
```

Comment:
just thinking whether it would be helpful if we list the env vars in the code as a variable or function instead of doc string?

Reply:
tbh, I don’t really like this parsing either, but I couldn’t come up with a better approach. Could you elaborate, please? How can we list it as “a variable or function”?

The only idea I have is to add `get_required_env` function to each provider and call it in `main.py`

Reply:
what do you think if we extend the abstract base class? we could do the check but the documenting (via `mkdocs`) would need some playing around with

here's an example in case i'm not explaining very well 😅 

```python3
class Provider(ABC):
    REQUIRED_ENV_VARS: list[str] = []

    @classmethod
    def check_env_vars(cls):
       ...

    def __init__(self) -> None:
        self.check_env_vars()

class BedrockProvider(Provider):
    REQUIRED_ENV_VARS = [
        "AWS_ACCESS_KEY_ID",
        "AWS_SECRET_ACCESS_KEY",
        "AWS_SESSION_TOKEN", 
    ]
```

then to get it into the docs, we could selectively generate it for one constant

```md
::: example.BedrockProvider
    options:
      members:
        - REQUIRED_ENV_VARS
```

but the formatting isn't the most readable and i'm not sure `mkdocstrings` gives us control of how to render `members`. which leaves us with potentially preprocessing to generate the docstring (maybe too complicated)

![image](https://github.com/user-attachments/assets/abac2204-ca77-4889-a862-3870d29a019c)